### ISSUE TITLE
Setting SuperUser for ADDS

### ISSUE THREAD

**Issue by zjorz** [neos-sdi/adfsmfa#148]
through the MMC, defining the super user as either domain\user or user@domain.com is not possible. It throws an error it is not valid. The same applies when defining the password

**Comment by redhook62** [comment]
hi **@zjorz** 

For all your issues (#147, #148, #149, #150), it's a security issue (ADMIN0120), 
this issue is not related with MFA.

Your can verify if your interactive Account has the right to open the ADFS management console. and with PowerShell 
`
Import-AdfsAuthenticationProviderConfigurationData -Name MultifactorAuthenticationProvider -FilePath c:\temp\mfa\adfsdump.xml
`
![image](https://user-images.githubusercontent.com/26826945/108625987-24e21700-744e-11eb-80bd-5ae4be3a7973.png)

As work around, we must search for this problem... 

But, if you connect to your server directly and not via RDP, it should do the job and everything seems working. 
After you can reconnect via RDP and it works.

let us know 
regards

**Comment by zjorz** [comment]
Everything is locally, no rdp and no remote management
Account i'm using is both a domain admin and an adfs admin. I can do anything on adfs server except for the issues mentioned
It is NOT a perms issue

**Comment by redhook62** [comment]
In this case we cannot reproduce your problems.

This morning, we test with a standard domainr user witch belongs only to the "ADFS Delegated Administration Group", not local administrator, not domain administrator and i can confirm to you, when a physical session (not RDP) have been succefully opened on the server, everything works like a charm with RDP also. 

it's a permission issue ! you need "Services administration Privileges"
regards

**Comment by zjorz** [comment]
these "access denied" messages of MFA only started after configured the so called SuperUser account through following steps

through the MMC it was impossible to set superuser for ADDS. Wanted to try through Powershell
through powershell did the following
$c = Get-MFASecurity
$c.Account = "DOMAIN\SUPERUSER"
$c.Password = "password"
$c

DeliveryWindow : 300
MaxRetries : 3
ReplayLevel : Full
LibVersion : V2
XORSecret : SOME_VALUE
PinLength : 4
DefaultPin : SOME_VALUE
DomainAddress : DOMAIN.COM
Account : DOMAIN\SUPERUSER
Password : password
SQLAccount :
SQLPassword :

Set-MFASecurity $c


WHAT is the purpose of the SuperUser in this case and is its config the reason for the Access Denied messages?

Because as my account is allowed to do anything on that at any level. I was also able to config the MFA extension before configuring the SuperUser Account. All the other issues already occurred before this access denied issue You can easily say you cannot reproduce it, but I'm experiencing issues,

The so called "Access Denied" issues. At first I was able to exec Get-MFAConfig and AFTER configuring the SuperUser that stopped working. So why and where is the access denied coming from

Are there any logs for the MFA extension I can have a look at?

**Comment by zjorz** [comment]
update....

deleted the "C:\Program Files\MFA\Config\system.db" file. Then I was able to open the MFA MMC and I got a new "C:\Program Files\MFA\Config\system.db" file and also in addition a "C:\Program Files\MFA\Config\config.db" file

**Comment by zjorz** [comment]
in the security configuration node I see the configured SuperUser account, but there are errors...

* for Domain Name it list DOMAIN.COM and it has a red exclamation mark saying "Invalid ADDS domain address!"
* for Access Account it list DOMAIN\ACCOUNT and it has a red exclamation mark saying "Invalid ADDS Account!"
* for Password it lists a value and it has a red exclamation mark saying "Invalid ADDS Password!"

Clicking on TEST it displays: "Unable to connect to ADDS forest/domain!"

What is this all about?
How should the domain name, the account and password be specified?

**Comment by redhook62** [comment]
Probably invalid domain or user !

**Comment by redhook62** [comment]
You can also remove "super user" informations, and see if it works

**Comment by zjorz** [comment]
no, the user is valid. it exists in AD and tried with RUNAS
please read the remarks I made. before even testing it is already giving errors that something is invalid. How can MFA know the password is invalid if it was not even tested? Just by specifying a value WITHOUT TESTING, it already states it is invalid. How is that even possible?

**Comment by redhook62** [comment]
Hi, 

also listen to what I tell you.

You have a problem accessing data on your domain, also consider looking at the eventlog.

I have  explained to you that it worked for a very long time (5 years ...). 
I have validated the operation on 24 different configurations, 
I have  also revalidated adding a standard Operator in the recommended groups, 
I have no problem on 2016 and 2012r2, for 2019 in ADFS SQL mode, I have explained the Work Around this morning. 

Test is made at each field update... perhaps not the better... 
If you click "TEST", it's a TEST, if the result is negative, review user informations, network, firewalls etc...

Your problems seem specific to your ADFS setup. 
Also check the attributes used in the Active Directory config, there are only basic LDAP requests in the code ... port 389 or 636
I can't help you anymore, if the test is negative consider it is true.

regards

redhook

**Comment by redhook62** [comment]
Strange thing !

Your domain is domain.com, but for import you are using OU=Users,OU=EMPLOYEES,OU=Org-Users,DC=XXXXXX,DC=**NET**
Ok, it' possible that you have multiple **toplevelnames**, the component manage this by default. 
if you have subdomains the component manage also this, and if you have valid trustrelationships the component also manage this.
Make sure taht your users have UPN

regards

**Comment by zjorz** [comment]
no, I only have one domain. the DOMAIN.COM is just some placeholder I just to not show my real domain on a public accessible forum. My real domain is <SOMETHING>.NET
I'm a bit surprised about your behavior. The only thing I see is that you are always pointing to somebody else and without knowing all the facts you make assumptions. Instead of continuously making unfounded statements, you could maybe ask questions to better understand the situation and determine together, where the real problem is. I truly do not care where it is, but rather prefer to solve a problem whatever or wherever it is.  I asked multiple questions which remain unanswered. Sure, I can also make mistakes, but until now in my AD and ADFS and other MFA providers are working perfectly, except this MFA extension. You mention it has been working for 5 years. I'm not going to say that is not true, nor is it my intention to say that. But please do not forget, that somebody else had issues with remote management. I also mentioned similar issues, while being locally on the server. Again you made all kinds of assumptions about my setup and basically telling it was my fault. In the end it was a bug in the software (I think powershell cmdlets). What surprises me, is the release of a piece of software for which the basic things (installing followed by configuring) were not working.
Now let's restart...
I found this product and thought it was interesting to use, hence why I gave it a try. I found issues and truly had difficulties troubleshooting it as I could not find anything telling what could be wrong, hence again why I raised an issue.
I want to be collaborative and try to solve this. For that I offered you to have a look with me in my test environment so that I better can explain the scenario. If you are not interested, that is fine by me and it would be a missed opportunity to improve the product if it had an underlying bug. So, please me know if you are interested or. It will not be a session for hours. 30 mins is more than enough. And again, this is not about getting support from you, but rather showing you what goes wrong with an explanation instead of trying to put everything in words
So, let me know if you interested or

**Comment by redhook62** [comment]
Hi @zjorz 

Hi,

Thank you for this return, which is hardly constructive.

You don't have to use this component, it's up to you. 
Do not invoke multiple and recurring bugs because this is not the case, at worst take the previous version in the meantime. 
The version number has changed, so there have been very important evolutions and obviously the bugs are inherent in this type of situation.
However, we did a battery of important tests,
the bug on the initial creation of a configuration in PowerShell was true (yes! we did not test that), but technically it was a detail.
The "SuperUser" for ADDS has been around for a very long time, and has also benefited from positive feedback in the past which has been incorporated.
I will switch these Issues to discussions, for me the bug is not reprodutible.
I'll give you some test items to move forward, and we'll see.

regards
